enterprise/frontend/src/metabase-enterprise/license/components/LicenseAndBillingSettings/LicenseAndBillingSettings.unit.spec.tsx

```diff
@@ -57,20 +60,140 @@ describe("LicenseAndBilling", () => {
+      value: "Manage preferences",
+      format: "string",
+      display: "external-link",
+      link: "https://store.metabase.com/",
+    };
+    const mockData: BillingInfo = {
+      version: "v1",
+      content: [
+        plan,
+        users,
+        nextCharge,
+        billingFreq,
+        nextChargeValue,
+        float,
+        unsupportedFormat,
+        unsupportedDisplay,
+        invalidValue,
+        managePreferences,
+      ],
+    };
+
+    fetchMock.get("path:/api/ee/billing", mockData);
+
     renderWithProviders(
-      <LicenseAndBillingSettings />,
-      setupState({ token: "token" }),
+      <Route path="/" component={LicenseAndBillingSettings}></Route>,
+      { withRouter: true, ...setupState({ token: "token" }) },
+    );
+
+    // test string format
+    expect(await screen.findByText(plan.name)).toBeInTheDocument();
+    expect(await screen.findByText(plan.name)).toBeInTheDocument();
+
+    // test integer format + internal-link display
+    expect(await screen.findByText(users.name)).toBeInTheDocument();
+    const userTableValue = await screen.findByTestId(
+      `billing-info-value-${getBillingInfoId(users)}`,
     );
+    expect(userTableValue).toHaveTextContent("4,000");
+    expect(userTableValue).toHaveAttribute("href", "/admin/people");
 
+    // test datetime format
+    expect(await screen.findByText(nextCharge.name)).toBeInTheDocument();
     expect(
-      await screen.findByText(
-        "Manage your Cloud account, including billing preferences, in your Metabase Store account.",
-      ),
+      await screen.findByText(`Monday, January 22, 2024`),
     ).toBeInTheDocument();
-    expect(screen.getByText("Go to the Metabase Store")).toBeInTheDocument();
+
+    // test currency
+    expect(await screen.findByText(nextChargeValue.name)).toBeInTheDocument();
+    expect(await screen.findByText(`$500.00`)).toBeInTheDocument();
+
+    // test float
+    expect(await screen.findByText(float.name)).toBeInTheDocument();
+    expect(screen.queryByText("" + float.value)).not.toBeInTheDocument();
+    expect(await screen.findByText("3.14")).toBeInTheDocument();
+
+    // test internal + external-link displays
+    expect(await screen.findByText(managePreferences.name)).toBeInTheDocument();
+    const managePreferencesTableValue = await screen.findByTestId(
+      `billing-info-value-${getBillingInfoId(managePreferences)}`,
+    );
+    expect(managePreferencesTableValue).toHaveTextContent(
+      managePreferences.value,
+    );
+    expect(managePreferencesTableValue).toHaveAttribute(
+      "href",
+      managePreferences.link,
+    );
+
+    // test unsupported display, unsupported format, and invalid items do not render
+    expect(screen.queryByText(unsupportedDisplay.name)).not.toBeInTheDocument();
+    expect(screen.queryByText(unsupportedFormat.name)).not.toBeInTheDocument();
+    expect(screen.queryByText(invalidValue.name)).not.toBeInTheDocument();
```

Comment:
I would maybe break these out into a separate test to make failures easier to figure out.